pynecone/components/graphing/victory.py

```diff
@@ -0,0 +1,567 @@
+        return format_xy(x, y)
+    else:
+        if len(x) != len(y) or len(x) != len(label):
+            raise ValueError("x, y, and label must be the same length")
+        return [{"x": x[i], "y": y[i], "label": label[i]} for i in range(len(x))]
+
+
+def format_voronoi(x: List, y: List) -> List:
+    """Format voronoi data.
+
+    Args:
+        x: The x values.
+        y: The y values.
+
+    Returns:
+        The formatted data.
+
+    Raises:
+        ValueError: If x or y is not provided.
+        ValueError: If x and y are not the same length.
+    """
+    data = []
+    if x is None or y is None:
+        raise ValueError("x and y must be specified")
+
+    return format_xy(x, y)
+
+
+def format_candlestick(x: List, open: List, close: List, high: List, low: List) -> List:
+    """Format candlestick data.
+
+    Args:
+        x: The x values.
+        open: The open values.
+        close: The close values.
+        high: The high values.
+        low: The low values.
+
+    Returns:
+        The formatted data.
+
+    Raises:
+        ValueError: If x is not provided.
+        ValueError: If x, open, close, high, and low are not the same length.
+    """
+    if x is None:
+        raise ValueError("x must be specified")
+
+    if (
+        len(x) != len(open)
+        or len(x) != len(close)
+        or len(x) != len(high)
+        or len(x) != len(low)
+    ):
+        raise ValueError("x, open, close, high, and low must be the same length")
+
+    return [
+        {"x": x[i], "open": open[i], "close": close[i], "high": high[i], "low": low[i]}
+        for i in range(len(x))
+    ]
+
+
+def format_error_bar(x: List, y: List, error_x: List, error_y: List) -> List:
+    """Format error bar data.
+
+    Args:
+        x: The x values.
+        y: The y values.
+        error_x: The error_x values.
+        error_y: The error_y values.
+
+    Returns:
+        The formatted data.
+
+    Raises:
+        ValueError: If x is not provided.
+        ValueError: If x, y, error_x, and error_y are not the same length.
+    """
+    if x is None:
+        raise ValueError("x must be specified")
+
+    if len(x) != len(error_x) or len(x) != len(error_y):
+        raise ValueError("x, y, error_x, and error_y must be the same length")
+    else:
+        return [
+            {"x": x[i], "y": y[i], "error_x": error_x[i], "error_y": error_y[i]}
+            for i in range(len(x))
+        ]
+
+
+def data(graph: str, x: List, y: Optional[List] = None, **kwargs) -> List:
+    """Format data.
+
+    Args:
+        graph: The graph type.
+        x: The x values.
+        y: The y values.
+        kwargs: The keyword arguments.
+
+    Returns:
+        The formatted data.
+
+    Raises:
+        ValueError: If graph is not provided.
+        ValueError: If graph is not supported.
+    """
+    if graph == "box_plot":
+        return format_box_plot(x, y, **kwargs)
+    elif graph == "bar":
+        return format_bar(x, y)  # type: ignore
```

Comment:
Why do these need the type ignores?

Reply:
The reason is sometimes y is not provide and was getting a None type error